internal/hcsoci/drivers.go

```diff
@@ -0,0 +1,109 @@
+// +build windows
+
+package hcsoci
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/oci"
+	"github.com/Microsoft/hcsshim/internal/uvm"
+	"github.com/pkg/errors"
+)
+
+// getAssignedDeviceUtilityTool gets the path to device-util on the server host
+func getAssignedDeviceUtilityTool(coi *createOptionsInternal) (string, error) {
+	tools, ok := coi.Spec.Annotations[oci.AnnotationAssignedDeviceUtilityTool]
+	if !ok || tools == "" {
+		return "", fmt.Errorf("no driver tools were specified %s", tools)
```

Comment:
Is the format directive necessary? Shouldn't tools in both cases be an empty string?

Reply:
In both cases? What do you mean? When a create call is issued, the spec will be updated with this annotation from the the config.toml, so this value could be present (and needs to be to avoid an error).

Reply:
Hmm maybe I'm missing something? If ok is false (E.g. the entry isn't in the map) then tools would be an empty string. If ok is true (E.g. the entry IS in the map) then tools would possibly not be an empty string, but in both of these if cases that you're checking tools should be an empty string.